src/test/java/com/bandwidth/sqs/consumer/SqsConsumerTest.java

```diff
@@ -123,6 +123,13 @@ public void testStartLongPollingRequest() {
         assertThat(requestCaptor.getValue()).isEqualTo(Optional.of(SqsConsumer.MAX_WAIT_TIME));
     }
 
+    @Test
+    public void testStopPrefetchIfPermitsZero() {
+        when(consumerHandlerMock.getPermitChangeRequests()).thenReturn(Observable.just(0));
+        consumer.start();
+        verify(sqsQueueMock, never()).receiveMessages(anyInt(), any(Duration.class));
```

Comment:
another unit test to test whether it will start consuming again if the num permits goes back up?